src/interfaces.ts

```diff
@@ -214,12 +214,18 @@ export type WithBrands<T, BrandNames extends string> = T & { readonly [brands]:
 export type Unbranded<T> = T extends WithBrands<infer Base, any> ? Base : T;
 
 /** Unbrand a given type (recursive). */
-export type DeepUnbranded<T> = T extends ReadonlyArray<unknown>
-    ? { [P in keyof T & number]: DeepUnbranded<T[P]> }
+export type DeepUnbranded<T> = T extends readonly [unknown, ...unknown[]] | []
+    ? UnbrandValues<T>
+    : T extends Array<infer E>
+    ? Array<DeepUnbranded<E>>
+    : T extends ReadonlyArray<infer E>
+    ? ReadonlyArray<DeepUnbranded<E>>
```

Comment:
I know `coverage` isn't really a thing here, but as far as I can tell, some of these 'branches' aren't tested.

Reply:
You are absolutely right. That is why I introduced `expect-type` in the first place, to make tests like this easier. I created a new file `interfaces.test.ts`. Let me know what you think.